src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -1981,8 +1981,13 @@ def get_multisite_status(self):
             is_multisite_configured = False
         return is_multisite_configured
 
-    def get_multisite_sync_status(self):
+    def get_multisite_sync_status(self, daemon_name: str):
         rgw_multisite_sync_status_cmd = ['sync', 'status']
+        daemons = _get_daemons()
+        if daemon_name:
+            realm_name = daemons[daemon_name].realm_name
```

Comment:
@afreen23 , `daemons` here is of type `RgwDaemon` and it always has a realm_name key. IMO get should only be used if we have any doubts if the key exists or not in the dict

Reply:
That's completly fine @afreen23, ideally we should use `get()` when we want to set a default value in case it fails to find the key in the dict. Otherwise we use the pythonic way to access a dictionary with the key: `dict[key]`. 
This way we'll be able to follow the principle, ask for forgiveness not permission, by just encapsulating the dict catching the `KeyError` exception.